### PR TITLE
Feature/10 custom rpcs

### LINKED ISSUE #10: Advanced user settings - RPC override
**User Story**
As a power user I want to be able to set my own RPC for each chain 

**Alternative Solutions**
Access from account modal, need to add the new advanced settings button which opens a new modal. 

We should then give an input to override the default RPC using a link provided by the user. We should check the availability of this link using a simple query before allowing the user to save. This option should then be saved specifically to the user's browser or local storage so it persists. 

An additional feature we can add here is the ability to override the app's decision to use the default data fetching option. In the case of guilds this should be a toggle between automatic and fallback (RPC only) where the user can choose to move from the default automatic logic to only using fallback data fetching. 
This choice should also be stored locally in browser.

### PR DESCRIPTION
**Pull request by rulfo71**
# Description

Adding the posibility of using Custom RPCs and decentralize mode: 

image

image
Closes #10 

- By leaving the text input blank and saving the user will go back to use davi's provider. 
- The URL provided is validated by checking it is valid and the chain provided is the correct one.
- After closing the modal the page will refresh entirely to modify the wagmi providers (`davi/src/provider/providers.ts`) with the changes in the modal. not sure if this the best approach but couldn't think of other... suggestions accepted 💡! 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

To test this i had to disable subgraph, which i did with the following: 

- `stores/modules/1_5/checkDataSourceAvailability` --> changed from `true` to `false`
- `stores/modules/guilds/SnapshotERC20Guild/index.ts` --> changed `checkDataSourceAvailability`  to: 
```
    checkDataSourceAvailability: () => {
      return Promise.resolve(false);
    },
```

Besides that, i tested adding as custom RPCs the ones we have on `providers.ts`: 
```
const POKT_NETWORK_URLS = {
  '1': 'https://eth-mainnet.gateway.pokt.network/v1/lb/dda01e253305bbeac6507a80',
  '5': 'https://eth-goerli.gateway.pokt.network/v1/lb/dda01e253305bbeac6507a80',
  '100': 'https://poa-xdai.gateway.pokt.network/v1/lb/dda01e253305bbeac6507a80',
};
```

switching them between correct and incorrect chains, removing parts of the string to check the error to show. 
Also tried inserting a simple text that wasn't a url. 

###Decentralize Mode 
test a subgraph function like `permissions` tab. This should not be working when `decentralize mode` is active

# Checklist:

- [x] My code follows the existing style of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any UI changes have been tested and made responsive for mobile views

### CODE REVIEW

**Comment by rossneilson** [comment]
Works great, loaded up my local node and saw it getting called in network tab.

A couple of things:
- Could we add the toggle for using default or custom RPC, can be a bit confusing knowing the user has to enter nothing to return to default
- Could we add the toggle for always using fallback/rpc from store? (just asked @borisblock3007 to add it to the designs also)
- I am seeing translation issues in vercel, but could just be caching maybe

**Comment by rulfo71** [comment]
>Could we add the toggle for always using fallback/rpc from store? (just asked @borisblock3007 to add it to the designs also)
@rossneilson By this you mean like the 'decentralize mode' we talked about?

**Comment by rossneilson** [comment]
> > Could we add the toggle for always using fallback/rpc from store? (just asked @borisblock3007 to add it to the designs also)
> > @rossneilson By this you mean like the 'decentralize mode' we talked about?

Yeah, actually a good name for it, would just be overriding our fallback logic

**Comment by rulfo71** [comment]
great! 
yes i heard that name from diogo, looks like carrot is implementing the same thing of optionally avoid using subgraph.

**Comment by rulfo71** [comment]
I added decentralize mode: 
image

i don't think this is the place we should add it, i just wanted to add the functionality until we have the designs.
Having it here is a bit confusing, first because this is a modal to change to customs RPCs which is not what we do with decentralize mode and also because i think it should be on a place where it is almost all the time visible so that the user can realize that a functionality is not working because of the decentralize mode is on. 

@borisblock3007 @rossneilson let me know what you think

**Comment by rossneilson** [comment]
> I added decentralize mode: image
> 
> i don't think this is the place we should add it, i just wanted to add the functionality until we have the designs. Having it here is a bit confusing, first because this is a modal to change to customs RPCs which is not what we do with decentralize mode and also because i think it should be on a place where it is almost all the time visible so that the user can realize that a functionality is not working because of the decentralize mode is on.
> 
> @borisblock3007 @rossneilson let me know what you think

I don't think its awful in here since it is related to RPCs. So we can either rename the RPC modal to "RPC configuration" or move decentralise mode into the main modal before RPC stuff. I don't have a huge opinion, lets choose one and we can move it later. 

We may even need to think about having the decentralise button be more a menu to select for each dao type.

**Comment by rossneilson** [comment]
Also the decentralize button works super well!!! great work

Small aside but seems the toggle button is sticky, if you move your cursor around it kind of follows (seems like smartphone behaviour but cant seem to make it not sticky again)

**Comment by rulfo71** [comment]
oh yes i'm seeing if you click on the black circle inside toggle that happens but if you click on the rest no... doesn't seem to happen on the other toggles, will look into it

**Comment by rulfo71** [comment]
I moved `DecentralizeMode` to the `WalletModal`. 
I think this is better because `CustomRPC` modal `onDismiss` refreshes the page and for `DecentralizeMode` we don't need that. 

image

File: apps/davi/public/locales/en/translation.json [hunk 1]
```diff
@@ -14,6 +14,21 @@
...
+    "saveChanges": "Save Changes",
+    "genericError": "Seems to be a problem with the url provided",
+    "wrongChain": "The url provided is not from chain {{chainName}}",
+    "decentralizeMode": "Decentralize mode",
+    "decentralizeModeTooltip": "This will disable subgraph and only use RPCs. Some functions will be lost"
+  },
+  "decentralizeMode": {
+    "decentralizeMode": "Decentralize mode",
+    "decentralizeModeTooltip": "This will disable subgraph and only use RPCs. Some functions will be lost"
+  },
```
Comment (rossneilson) [inline apps/davi/public/locales/en/translation.json:31]: We'll need to consider how we do translation work from now on. For now you could add the Spanish or make another task to translate the customRPC object

**Review by rossneilson — APPROVED** [review]

File: apps/davi/public/locales/en/translation.json (hunk 1 above)
Comment (rulfo71) [reply apps/davi/public/locales/en/translation.json:31]: sure, i'll make another task so that we get this pr merged

**Pull request merged**